src/common/uilocale.cpp

```diff
@@ -240,10 +467,31 @@ bool wxUILocale::UseDefault()
 }
 
 /* static */
-bool wxUILocale::UseLanguage(const wxLanguageInfo& info)
+bool wxUILocale::UseLocaleName(const wxString& localeName)
 {
-    wxUILocaleImpl* const impl = wxUILocaleImpl::CreateForLanguage(info);
-    if ( !impl )
+    wxUILocaleImpl* impl = NULL;
+    if (localeName.IsSameAs("C", false))
+    {
+        impl = wxUILocaleImpl::CreateStdC();
+    }
+    else
+    {
+        wxLocaleIdent localeId = wxLocaleIdent::FromTag(localeName);
+        impl = wxUILocaleImpl::CreateForLocale(localeId);
+        if (!impl)
+        {
+            // Creating the locale may have failed due to lacking support for wxUILocaleImplName
```

Comment:
Sorry, what does this comment mean?

Reply:
Method `wxUILocaleImplName::CanUse()` that is called internally by method `wxUILocaleImpl::CreateForLocale()` on Windows platforms can potentially fail due to missing Windows system calls. In that case, a different approach for creating a wxUILocale instance (namely via LCID) is tested. This should affect only applications running on older Windows versions (like Windows XP).